src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
Note that I hadn't actually concluded anything after my discussion w/ @daxian-dbw, but after some of my own experiments, I do agree. The existing behavior is surprising and not useful, anyone potentially hit by the breaking change wasn't getting what they wanted anyway.